agent/rpc/middleware/interceptors.go

```diff
@@ -33,33 +33,79 @@ var OneTwelveRPCSummary = []prometheus.SummaryDefinition{
 }
 
 type RequestRecorder struct {
-	Logger       hclog.Logger
-	RecorderFunc func(key []string, val float32, labels []metrics.Label)
+	Logger         hclog.Logger
+	RecorderFunc   func(key []string, val float32, labels []metrics.Label)
+	serverIsLeader func() bool
+	localDC        string
 }
 
-func NewRequestRecorder(logger hclog.Logger) *RequestRecorder {
-	return &RequestRecorder{Logger: logger, RecorderFunc: metrics.AddSampleWithLabels}
+func NewRequestRecorder(logger hclog.Logger, isLeader func() bool, localDC string) *RequestRecorder {
+	return &RequestRecorder{
+		Logger:         logger,
+		RecorderFunc:   metrics.AddSampleWithLabels,
+		serverIsLeader: isLeader,
+		localDC:        localDC,
+	}
 }
 
 func (r *RequestRecorder) Record(requestName string, rpcType string, start time.Time, request interface{}, respErrored bool) {
 	elapsed := time.Since(start).Milliseconds()
 	reqType := requestType(request)
+	serverRole := r.getServerRole()
 
 	labels := []metrics.Label{
 		{Name: "method", Value: requestName},
 		{Name: "errored", Value: strconv.FormatBool(respErrored)},
 		{Name: "request_type", Value: reqType},
 		{Name: "rpc_type", Value: rpcType},
+		{Name: "server_role", Value: serverRole},
```

Comment:
nit/bikeshed:

```suggestion
		{Name: "leader", Value: strconv.FormatBool(isLeader)},
```

RPC metrics will only be emitted by servers so there is no need to prefix the metric name with `server`. Secondly since there are only two possible values of leader and follower I think the metric might be more intuitive for someone exploring their grafana or prometheus to have the name be `leader` and the value be a boolean.

Reply:
this makes a lot of sense!  and that's what I am changing for a new commit ⌨️ 

I think it would be nice to keep `unreported`, and subsequently keep this as a tri state 

```
leader: {`true`, `false`, `unreported`}
```

for a defense in depth kind of approach. If the function is not passed to the `RequestRecorder`, I wouldn't think it useful or accurate to report `leader: false`